tests/test_core.py

```diff
@@ -634,3 +634,11 @@ def test_array_creation_with_invalid_fields():
 def test_promote_nullable():
     with pytest.warns(DeprecationWarning):
         assert ndx.promote_nullable(np.int64) == ndx.nint64
+
+
+@pytest.mark.parametrize("val", [1, 1.0, 1.5, 0.123456789, "a"])
```

Comment:
```suggestion
@pytest.mark.parametrize("val", [1, 1.0, 1.5, 0.123456789, "a", np.float64(0.123456789)])
```

Reply:
I think this will still be broken. `min_scalar_type` seems to have odd semantics.

Reply:
Can you please create a separate branch. I don't quite understand your comment. This PR explicitly doesn't use `np.min_scalar_type` for python float scalars.

Reply:
Turns out that `np.float64` objects are instances of `np.generic` and `float`. Thus `np.min_scalar_type` is indeed not used, event though I'd say it is not overly explicit. If it had been used, we would have seen the following sematics that I do find very surprising:
```
np.min_scalar_type(np.float64(0.123456789))
>>> dtype('float16')
```

Reply:
Yeah this definitely warrants an upstream issue in NumPy.

Reply:
Yup totally, it is not obvious what is going on for floats inside `min_scalar_type` - more context is in https://github.com/numpy/numpy/issues/8207.